applications/RANSApplication/custom_strategies/generic_convergence_criteria.h

```diff
@@ -0,0 +1,213 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Suneth Warnakulasuriya
+//
+
+#ifndef KRATOS_GENERIC_CONVERGENCE_CRITERIA_H
+#define KRATOS_GENERIC_CONVERGENCE_CRITERIA_H
+
+// System includes
+#include <cmath>
+#include <string>
+#include <unordered_map>
+
+/* Project includes */
+#include "includes/model_part.h"
+#include "solving_strategies/convergencecriterias/convergence_criteria.h"
+
+namespace Kratos
+{
+///@addtogroup IncompressibleFluidApplication
+///@{
+
+///@name Kratos Classes
+///@{
+
+/// Convergence criteria for fluid problems.
+/**
+ This class implements a convergence control based on nodal velocity and
+ pressure values. The error is evaluated separately for each of them, and
+ relative and absolute tolerances for both must be specified.
+ */
+template <class TSparseSpace, class TDenseSpace>
+class KRATOS_API(RANS_APPLICATION) GenericConvergenceCriteria
+    : public ConvergenceCriteria<TSparseSpace, TDenseSpace>
```

Comment:
May this class be substituted by the KratosCore one proposed in #6861?

Reply:
yes, I plan to substitue it when #6861 is merged.